solutions/silver/cf-1000C.mdx

```diff
@@ -63,31 +62,181 @@ int main() {
+		for (int i = 0; i < n; i++) {
+			StringTokenizer st = new StringTokenizer(br.readLine());
+			segments[i] = new Pair(Long.valueOf(st.nextToken()),
+			                       Long.valueOf(st.nextToken()));
+			// In a set, only unique points will be stored
+			points.add(segments[i].first);
+			points.add(segments[i].second + 1);
+		}
+		br.close();
+
+		int cur = 0;
+		// This map stores the compressed coordinates.
+		Map<Long, Integer> compressed = new HashMap<>();
+		List<Long> coords = new ArrayList<>();
+		for (long c : points) {
+			// Assign c to cur, where cur is our compressed coordinate for "a".
+			compressed.put(c, cur);
+			/*
+			 * We still need to remember our original endpoints so
+			 * we can retransform the compressed coordinates
+			 */
+			coords.add(c);
+			cur++;
+		}
+		// This stores the frequency of a given endpoint.
+		int[] freq = new int[2 * n];
+		for (int i = 0; i < n; i++) {
+			/*
+			 * A segment starts from [l*, r*], so we should end the segment
+			 * at r* + 1 to ensure that r* is included into the segment.
+			 */
+			freq[compressed.get(segments[i].first)]++;
+			freq[compressed.get(segments[i].second + 1)]--;
+		}
+		// prefix sums for number of points at a given compressed endpoint
+		for (int i = 1; i < 2 * n; i++) { freq[i] += freq[i - 1]; }
+
+		// cnt[i] := number of points covered by exactly i segments
+		long[] cnt = new long[n + 1];
+		for (int i = 1; i < coords.size(); i++) {
+			/*
+			 * We decompress the endpoints for a given frequency and add the
+			 * points in this range to the answer.
+			 */
+			cnt[freq[i - 1]] += coords.get(i) - coords.get(i - 1);
+		}
+
+		for (int i = 1; i <= n; i++) {
+			System.out.print(cnt[i]);
+			System.out.print(" ");
+		}
+	}
+
+	// BeginCodeSnip{class Pair}
+	private static class Pair {
+		public long first, second;
+
+		public Pair(long f, long s) {
+			first = f;
+			second = s;
+		}
+	}
+	// EndCodeSnip
+}
+```
+
+</JavaSection>
+
+</LanguageSection>
+
+
+## Solution 2: Map
+
+Alternatively, we can also use a map to store frequencies at each point of
+interest and iterate through them in order. This way, we don't need to
+explicitly perform the coordinate compression.
+
+### Implementation
+
+**Time Complexity:** $\mathcal{O}(N\log N)$
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+using ll = long long;
+
+int main() {
+	int n;
+	cin >> n;
+
+	// distinct coordinates of the start and end of the segmetns
+	set<ll> points;
+	// frequency of a given coordinate
+	map<ll, int> freq;
+	for (int i = 0; i < n; i++) {
+		ll l, r;
+		cin >> l >> r;
+
+		/*
+		 * A segment starts from [l*, r*], so we should end the segment
+		 * at r* + 1 to ensure that r* is included into the segment.
+		 */
+		freq[l]++;
+		freq[r + 1]--;
+		points.insert(l);
+		points.insert(r + 1);
+	}
+
+	// number of overlapping segments at current position
+	int layers = freq[*points.begin()];
+	auto it = points.begin();
+	it++;
```

Comment:
```suggestion
	auto it = points.begin()++;
```
does this work? not sure
also i'm sure there's a reason but can you tell me why you start at the 1st pt instead of the 0th

Reply:
nope this doesn't work, but `next()` does though

it's not the first cause we always look backwards to determine the # of overlapping segments and the # of points, therefore we start with the range [0;1]